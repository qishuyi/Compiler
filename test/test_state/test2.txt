let x : <int> = ref 5 in
  while !x > 1 do x := !x -1 end ; !x
