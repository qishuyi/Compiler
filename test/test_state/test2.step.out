(let x = (ref 5) in (while (> !x 1) do (x := (- !x 1)) end) ; !x)
(let x = Ptr(0) in (while (> !x 1) do (x := (- !x 1)) end) ; !x)
(while (> !Ptr(0) 1) do (Ptr(0) := (- !Ptr(0) 1)) end) ; !Ptr(0)
(while (> !Ptr(0) 1) do (Ptr(0) := (- !Ptr(0) 1)) end) ; !Ptr(0)
(while (> !Ptr(0) 1) do (Ptr(0) := (- !Ptr(0) 1)) end) ; !Ptr(0)
(while (> !Ptr(0) 1) do (Ptr(0) := (- !Ptr(0) 1)) end) ; !Ptr(0)
(while (> !Ptr(0) 1) do (Ptr(0) := (- !Ptr(0) 1)) end) ; !Ptr(0)
() ; !Ptr(0)
() ; 1
1
